Zend/zend_ast.c

```diff
@@ -702,6 +702,71 @@ ZEND_API zend_result ZEND_FASTCALL zend_ast_evaluate_inner(
 			}
 			zval_ptr_dtor_nogc(&op1);
 			break;
+		case ZEND_AST_CAST:
+			if (UNEXPECTED(zend_ast_evaluate_ex(&op1, ast->child[0], scope, &short_circuited, ctx) != SUCCESS)) {
```

Comment:
Does this condition mean AST isn't compilable on compile time? If so, it should not be "UNEXPECTED" nor "EXPECTED" alrought this is minor.